mempool/mock/mempool.go

```diff
@@ -8,17 +8,31 @@ import (
 )
 
 // Mempool is an empty implementation of a Mempool, useful for testing.
-type Mempool struct{}
+type Mempool struct {
+	txs        map[types.TxKey]types.Tx
+	evictedTxs map[types.TxKey]bool
+}
```

Comment:
It might be better to use something like mockery and have generated mocks then to reuse this empty implementation. It also means that you've had to touch a lot of different files